ucc/tests/test_compile.py

```diff
@@ -131,6 +109,52 @@ def run(self, dag):
     assert_same_circuits(post_compiler_circuit, CirqCircuit(X(qubit)))
 
 
+@pytest.mark.parametrize("check", ["opset", "coupling_map"])
+def test_compile_with_target_device(check):
```

Comment:
I've never seen this pattern before. I would think it would be more natural to have two tests `test_target_device_gateset_respected` and `test_target_device_coupling_map_respected` (modulo naming). Or even just removing the parametrization and the `if`/`elif` blocks and having two asserts. That does technically go against the "only `assert` one thing per test" rule, but IMO the assertion is that the `target_device` is being respected and that takes two assertions.